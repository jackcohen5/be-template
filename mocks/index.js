import { Roles } from 'functions/BaseView'

// COMMON
export const mockUserId1 = 'user_id_1'

// LAMBDA
export const makeAuthenticatedEvent = ({
    roles = [Roles.TEMPLATE_NAME_ROLE1],
} = {}) => ({
    requestContext: { authorizer: { userId: mockUserId1, roles } },
})

export const authenticatedEvent = makeAuthenticatedEvent()

// ITEM
export const mockItem1 = {
    itemId: 'item_id_1',
    userId: mockUserId1,
}

export const mockItem2 = {
    itemId: 'item_id_2',
    userId: 'user_id_2',
}

export const mockItems = [mockItem1, mockItem2]
