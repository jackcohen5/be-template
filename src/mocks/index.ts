import {
    APIGatewayEventRequestContext,
    APIGatewayProxyEvent,
    APIGatewayProxyEventBase,
    APIGatewayProxyEventPathParameters,
} from 'aws-lambda'

import { Roles } from 'functions/authorizer/authorizer.constants'

// COMMON
export const mockUserId1 = 'user_id_1'

// LAMBDA
const baseEvent: APIGatewayProxyEvent = {
    body: '',
    headers: undefined,
    multiValueHeaders: undefined,
    httpMethod: '',
    isBase64Encoded: false,
    path: '',
    pathParameters: undefined,
    queryStringParameters: undefined,
    multiValueQueryStringParameters: undefined,
    stageVariables: undefined,
    requestContext: undefined,
    resource: '',
}

const baseResourceContext: APIGatewayEventRequestContext = {
    accountId: '',
    apiId: '',
    authorizer: undefined,
    protocol: '',
    httpMethod: '',
    identity: undefined,
    path: '',
    stage: '',
    requestId: '',
    requestTimeEpoch: 0,
    resourceId: '',
    resourcePath: '',
}

export const makeEvent = <TAuthorizerContext>({
    authorizer,
    body = '{}',
    pathParameters = {},
}: {
    body?: string
    pathParameters?: APIGatewayProxyEventPathParameters
    authorizer?: TAuthorizerContext
} = {}): APIGatewayProxyEventBase<TAuthorizerContext> => ({
    ...baseEvent,
    requestContext: {
        ...baseResourceContext,
        authorizer,
    },
    body: JSON.stringify(body),
    pathParameters,
})

export const makeAuthenticatedEvent = ({
    body,
    pathParameters,
    roles = [Roles.TEMPLATE_NAME_ROLE1],
}: {
    body?: string
    pathParameters?: APIGatewayProxyEventPathParameters
    roles?: Roles[]
} = {}): APIGatewayProxyEvent =>
    makeEvent({
        authorizer: { userId: mockUserId1, roles },
        body,
        pathParameters,
    })

export const role1AuthenticatedEvent = makeAuthenticatedEvent()

// ITEM
export const mockItem1 = {
    itemId: 'item_id_1',
    userId: mockUserId1,
}

export const mockItem2 = {
    itemId: 'item_id_2',
    userId: 'user_id_2',
}

export const mockItems = [mockItem1, mockItem2]
