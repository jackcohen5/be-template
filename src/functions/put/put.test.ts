import { role1AuthenticatedEvent, role2AuthenticatedEvent } from 'mocks'

import PutAPIView from '.'

describe('put', () => {
    it('returns 403 for invalid role', async () => {
        const response = await PutAPIView(role2AuthenticatedEvent)
        expect(response.statusCode).toBe(403)
    })

    it('executes as expected', async () => {
        const response = await PutAPIView(role1AuthenticatedEvent)
        expect(response).toMatchSnapshot()
    })
})
