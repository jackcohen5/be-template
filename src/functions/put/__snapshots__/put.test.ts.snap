// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`put executes as expected 1`] = `
Object {
  "body": "{\\"item\\":{\\"itemId\\":\\"item_id_1\\",\\"userId\\":\\"user_id_1\\"}}",
  "headers": Object {
    "Access-Control-Allow-Origin": "*",
  },
  "statusCode": 200,
}
`;
